### PR TITLE
Docker image for running in ECS

### PR DESCRIPTION
**Pull request by amenasse**
Dockerfile for running any of the tupaia services. Due to issues with the monorepo tooling (which i will document elsewhere), it wasn't possible to build individual images for each service. 

The central-server is started by default. This can be customised by passing a different workdir at runtime:

    podman run  -it  --workdir /app/packages/admin-panel-server/  localhost/tupaia:latest

The dev dependencies are included in the resulting image, as attempting to remove them caused runtime deps to be removed :(

### CODE REVIEW

File: packages/devops/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,104 @@
+FROM node:14.19.3-alpine3.15 as builder
```
Comment (amenasse) [inline packages/devops/Dockerfile:1]: Node version matches engine in package.json, image matches whats used in codeship

File: packages/devops/Dockerfile [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+COPY packages/psss-server/package.json ./packages/psss-server/
+COPY packages/report-server/package.json ./packages/report-server/
+COPY packages/server-boilerplate/package.json ./packages/server-boilerplate/
+COPY packages/tsutils/package.json ./packages/tsutils/
+COPY packages/types/package.json ./packages/types/
+COPY packages/ui-components/package.json ./packages/ui-components/
+COPY packages/utils/package.json ./packages/utils/
+COPY packages/weather-api/package.json ./packages/weather-api/
+COPY packages/web-config-server/package.json ./packages/web-config-server/
+COPY packages/web-frontend/package.json ./packages/web-frontend/
```
Comment (passcod) [inline packages/devops/Dockerfile:60]: This is pretty annoying and also will create upwards of 30 layers for not much. At the very least using a multi dockerfile will help, but really you can use an intermediate step to take care of this programmatically, see this pattern: https://github.com/beyondessential/tamanu/blob/dev/Dockerfile.codeship#L18-L23

Comment (amenasse) [reply packages/devops/Dockerfile]: resolved by 8087a76a1ae8ff40ba71cdf8a3a81952138116e4

**Comment by amenasse** [comment]
After using it for a bit in the ECS prototype I think this is ready. I've removed the LastPass stuff for now, will address this in a separate PR

**Comment by amenasse** [comment]
Not sure why Codeship build is failing as i don't have visibility on tupaia in Codeship

**Comment by passcod** [comment]
@IgorNadj @rohan-bes Would be good to have your ✅ on this too :)

**Review by passcod — APPROVED** [review]

**Comment by IgorNadj** [comment]
> Not sure why Codeship build is failing as i don't have visibility on tupaia in Codeship

The error in Codeship is:
```
Build Error: (step: Validate_and_test_Validate_Validate_new_migrations) error loading services during run step: failure to build Image{ name: "validation", branch: "ant/docker-production", dockerfile: "/data/clone/packages/devops/ci/validation.Dockerfile", 
...
API error (500): Cannot locate specified Dockerfile: ./packages/devops/ci/validation.Dockerfile, (
```

It looks like adding `packages/devops` to the .dockerignore means the other images which rely on being able to copy stuff from it (configs/scripts) failed to build.

**Review by IgorNadj — APPROVED** [review]
LGTM! Thanks for that

**Comment by amenasse** [comment]
> 
> It looks like adding `packages/devops` to the .dockerignore means the other images which rely on being able to copy stuff from it (configs/scripts) failed to build. Have a look at packages/devops/ci/tupaia-ci-cd.Dockerfile and validation.Dockerfile

Thanks @IgorNadj, I addressed this in 4631b1a017e17ab403fa9b662a023bcb9f489d0d so hopefully it passes now :crossed_fingers:

**Comment by IgorNadj** [comment]
> > It looks like adding `packages/devops` to the .dockerignore means the other images which rely on being able to copy stuff from it (configs/scripts) failed to build. Have a look at packages/devops/ci/tupaia-ci-cd.Dockerfile and validation.Dockerfile
> 
> Thanks @IgorNadj, I addressed this in [4631b1a](https://github.com/beyondessential/tupaia/commit/4631b1a017e17ab403fa9b662a023bcb9f489d0d) so hopefully it passes now 🤞

That did fix it, but another thing sorry! Branch names can't have / in them. I think it's alphanumeric and -. Do you mind recreating this PR with the new branch name and I'll approve. Thanks!

**Comment by amenasse** [comment]
Closing in favour of #4519 (branch name fixed)

**Pull request closed without merging**